apps/dashboard/src/components/header-navigation/edit-bridge-url-button.tsx

```diff
@@ -0,0 +1,124 @@
+    setError,
+    formState: { isDirty, errors },
+  } = form;
+  const { currentEnvironment, setBridgeUrl } = useEnvironment();
+  const { status, bridgeURL: envBridgeUrl } = useBridgeHealthCheck();
+  const { validateBridgeUrl, isPending: isValidatingBridgeUrl } = useValidateBridgeUrl();
+  const { updateBridgeUrl, isPending: isUpdatingBridgeUrl } = useUpdateBridgeUrl();
+
+  useLayoutEffect(() => {
+    reset({ bridgeUrl: envBridgeUrl });
+  }, [reset, envBridgeUrl]);
+
+  const onSubmit = async ({ bridgeUrl }: z.infer<typeof formSchema>) => {
+    const { isValid } = await validateBridgeUrl(bridgeUrl);
+    if (isValid) {
+      await updateBridgeUrl({ url: bridgeUrl, environmentId: currentEnvironment?._id ?? '' });
+      setBridgeUrl(bridgeUrl);
+    } else {
+      setError('bridgeUrl', { message: 'The provided URL is not the Novu Endpoint URL' });
+    }
+  };
+
+  return (
+    <Popover
+      open={isPopoverOpen}
+      onOpenChange={(newIsOpen) => {
+        setIsPopoverOpen(newIsOpen);
+        if (!newIsOpen && isDirty) {
+          reset({ bridgeUrl: envBridgeUrl });
+        }
+      }}
+    >
+      <PopoverTrigger asChild>
+        <button className="text-foreground-600 flex h-6 items-center gap-2 rounded-md border border-neutral-200 text-xs leading-4 hover:bg-neutral-50 focus:bg-neutral-50">
+          <div className="flex items-center gap-2 px-1.5 py-1">
+            <span
+              className={cn(
+                'relative size-1.5 animate-[pulse-shadow_1s_ease-in-out_infinite] rounded-full',
+                status === ConnectionStatus.DISCONNECTED || status === ConnectionStatus.LOADING
+                  ? 'bg-destructive [--pulse-color:var(--destructive)]'
+                  : 'bg-success [--pulse-color:var(--destructive)]'
+              )}
+            />
+            <span>Local Studio</span>
+          </div>
+          <span className="border-l border-neutral-200 p-1.5">
+            <RiPencilFill className="size-[12px]" />
+          </span>
+        </button>
+      </PopoverTrigger>
+      <PopoverPortal>
+        <PopoverContent className="w-[362px] p-0" side="bottom" align="end">
+          <Form {...form}>
+            <form onSubmit={handleSubmit(onSubmit)}>
+              <div className="flex flex-col gap-1 p-5">
+                <FormField
+                  control={control}
+                  name="bridgeUrl"
+                  render={({ field }) => (
+                    <FormItem>
+                      <FormLabel>Bridge Endpoint URL</FormLabel>
+                      <FormControl>
+                        <InputField variant="xs" state={errors.bridgeUrl?.message ? 'error' : 'default'}>
+                          <RiLinkM className="size-5 min-w-5" />
+                          <Input id="bridgeUrl" {...field} />
+                        </InputField>
+                      </FormControl>
+                      <FormMessage>Full path URL (e.g., https://your.api.com/api/novu)</FormMessage>
```

Comment:
`Full path URL` doesn't make a lot of sense as an expression. Use `URL` or `Set URL`.

```suggestion
                      <FormMessage>URL (e.g., https://your.api.com/api/novu)</FormMessage>
```